### DISCUSSION TITLE
m2m relationships

### DISCUSSION THREAD

**Discussion by tr11** [aminalaee/sqladmin#91] [Q&A]
What's the recommended way to use sqladmin with many-to-many relationships with an association table?  I have the following models:

```
class User(Base):
    __tablename__ = "users"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String, index=True, nullable=True)
    groups = relationship("UserGroup", back_populates="user")

    def __repr__(self) -> str:
        return self.name

class Group(Base):
    __tablename__ = "groups"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String, unique=True, nullable=False)
    users = relationship("UserGroup", back_populates="group")

    def __repr__(self) -> str:
        return self.name

class UserGroup(Base):
    __tablename__ = "user_groups"
    __table_args__ = (UniqueConstraint("user_id", "group_id"),)

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(ForeignKey(User.id))
    group_id = Column(ForeignKey(Group.id))
    user = relationship(User, back_populates="groups")
    group = relationship(Group, back_populates="users")

    def __repr__(self) -> str:
        return f"{self.group} ({self.user})"
```

It seems that the methods executed from the HTML template do not execute inside a session, which causes errors such as 
```sqlalchemy.orm.exc.DetachedInstanceError: Parent instance <UserGroup at 0x7fdec0d3e7d0> is not bound to a Session; lazy load operation of attribute 'group' cannot proceed (Background on this error at: https://sqlalche.me/e/14/bhk3)```

Is there something I'm missing with m2m relationships?

**Comment by aminalaee** [comment]
Hey, I'm not really sure about this particular error but I think the m2m needs a bit of modifications in the forms.py area. I'll take a look into this over the weekend.

**Comment by jpribyl** [comment]
Awesome work on this package, thank you so much :pray: 

I am also interested in this! Are there any test cases / examples you could point me towards @aminalaee ?

**Reply by aminalaee** [reply]
@jpribyl Thanks for the kind words.

Just to be clear, with the latest code, running the M2M example with association table works fine: https://docs.sqlalchemy.org/en/14/orm/basic_relationships.html#many-to-many

And running the M2M with the Association Object shows empty forms which needs to be fixed to fetch related data. Can you please confirm this?

**Comment by jpribyl** [comment]
Yeah sorry I should have updated my comment :flushed: I realized shortly after that the issue was actually this one https://github.com/tiangolo/sqlmodel/issues/315

cheers @aminalaee 

btw- still up in the air but there's a chance I may wind up using this enough to throw a PR or two your way. Specifically may wind up wanting to add a search / text filter to it eventually. Also may try beefing up support for enums. It's not more than a mild inconvenience rn so we'll see!

**Comment by jpribyl** [comment]
Doh! I see this now- looks like maybe it's already got a search https://aminalaee.github.io/sqladmin/api_reference/model_admin/#sqladmin.models.ModelAdmin.column_searchable_list
